Formula/youtube-dl.rb

```diff
@@ -16,6 +16,11 @@ class YoutubeDl < Formula
 
   depends_on "python@3.9"
 
+  resource "youtube_dl" do
+    url "https://files.pythonhosted.org/packages/76/03/a5c749ea2ddeac51dcd0376d8519868ca710dd241c871f555765e777c5bf/youtube_dl-2020.11.21.1.tar.gz"
+    sha256 "a785c1373a3c2d0b82c54aabc4831e8e6f6ede059ec462e54526d694dd3c29ca"
+  end
+
```

Comment:
This doesn't look right

Reply:
I think the `youtube_dl` from the `pipgrip` output not matching `youtube-dl` makes the code think it's a new resource. cc @rylan12

Reply:
Ah, this is a bug in `update-python-resources`. According to [PEP 426](https://www.python.org/dev/peps/pep-0426/#name):

> All comparisons of distribution names MUST be case insensitive, and MUST consider hyphens and underscores to be equivalent.

So, `update-python-resources` should treat `youtube_dl` and `youtube-dl` as the same. I'll update the currently open `update-python-resources` PR to include this fix.